packages/filebrowser/src/opendialog.ts

```diff
@@ -125,57 +140,68 @@ class OpenDialog
     manager: IDocumentManager,
     filter?: (value: Contents.IModel) => Partial<IScore> | null,
     translator?: ITranslator,
+    defaultPath?: string,
     filterDirectories?: boolean
   ) {
     super();
     translator = translator ?? nullTranslator;
     const trans = translator.load('jupyterlab');
     this.addClass(OPEN_DIALOG_CLASS);
 
-    this._browser = Private.createFilteredFileBrowser(
+    Private.createFilteredFileBrowser(
       'filtered-file-browser-dialog',
       manager,
       filter,
       {},
       translator,
+      defaultPath,
       filterDirectories
-    );
+    )
+      .then(browser => {
+        this._browser = browser;
 
-    // Add toolbar items
-    setToolbar(this._browser, (browser: FileBrowser) => [
-      {
-        name: 'new-folder',
-        widget: new ToolbarButton({
-          icon: newFolderIcon,
-          onClick: () => {
-            void browser.createNewDirectory();
-          },
-          tooltip: trans.__('New Folder')
-        })
-      },
-      {
-        name: 'refresher',
-        widget: new ToolbarButton({
-          icon: refreshIcon,
-          onClick: () => {
-            browser.model.refresh().catch(reason => {
-              console.error(
-                'Failed to refresh file browser in open dialog.',
-                reason
-              );
-            });
+        // Add toolbar items
+        setToolbar(this._browser, (browser: FileBrowser) => [
+          {
+            name: 'new-folder',
+            widget: new ToolbarButton({
+              icon: newFolderIcon,
+              onClick: () => {
+                void browser.createNewDirectory();
+              },
+              tooltip: trans.__('New Folder')
+            })
           },
-          tooltip: trans.__('Refresh File List')
-        })
-      }
-    ]);
+          {
+            name: 'refresher',
+            widget: new ToolbarButton({
+              icon: refreshIcon,
+              onClick: () => {
+                browser.model.refresh().catch(reason => {
+                  console.error(
+                    'Failed to refresh file browser in open dialog.',
+                    reason
+                  );
+                });
+              },
+              tooltip: trans.__('Refresh File List')
+            })
+          }
+        ]);
+
+        // Build the sub widgets
+        const layout = new PanelLayout();
+        layout.addWidget(this._browser);
 
-    // Build the sub widgets
-    const layout = new PanelLayout();
-    layout.addWidget(this._browser);
+        // Set Widget content
+        this.layout = layout;
 
-    // Set Widget content
-    this.layout = layout;
+        this._ready.resolve();
+      })
+      .catch(() => {
+        console.error('Error while creating file browser in open dialog');
+        this._ready.reject(void 0);
```

Comment:
Wasn't sure how to approach the error handling. Would it be best to also pass an error message during the rejection?

Reply:
I do not have an opinion here. I think it is fine to merge as-is and only expand if someone asks for it. However, please look at a suggestion above to expose the failure reason in console for easier debugging.